numpy/lib/tests/test_index_tricks.py

```diff
@@ -262,6 +269,25 @@ def test_accepts_npfloating(self):
         assert_(grid32.dtype == np.float64)
         assert_array_almost_equal(grid64, grid32)
 
+    @pytest.mark.skipif(FLOAT128_EX, reason=str(FLOAT128_EX))
+    def test_accepts_npfloating128(self):
+        # regression tests for #16945
+        grid64 = mgrid[0.1:0.33:0.1,]
+        grid128 = mgrid[np.float128(0.1):np.float128(0.33):np.float128(0.1),]
+        assert_(grid128.dtype == np.float128)
+        assert_array_almost_equal(grid64, grid128)
+
+        grid128c_a = mgrid[0:np.float128(1):3.4j]
+        grid128c_b = mgrid[0:np.float128(1):3.4j,]
+        assert_(grid128c_a.dtype == grid128c_b.dtype == np.float128)
+        assert_array_almost_equal(grid128c_a, grid128c_b[0])
```

Comment:
I'd expect these to be exactly equal, not just almost equal

Reply:
I feel it would be wise to stick to `assert_array_almost_equal()` considering we're dealing with float arrays.
just in case to deal with any numerical noise that might creep in.

Reply:
"noise" does not creep into deterministic calculations like this - computing a 1d vs 2d grid of numbers should result in identical bitpatterns, else we have another bug.

Reply:
Ah yes, I missed the fact that this comparison involved two arrays of the same dtype.
Addressed in https://github.com/numpy/numpy/pull/16971/commits/2e94e7e9f54efb17be9a59f14f55deae1e6e758b.